FrontEnd/src/Header.jsx

```diff
@@ -1,7 +1,36 @@
 import { NavLink } from "react-router-dom";
-import {useState} from "react"
+import { useEffect, useState } from "react";
 import "./Header.css";
-function Header({ userData }) {
+import NotificationsPage from "./Notifications/NotificationsPage";
+function Header({ userData, socket}) {
+  const [notifications, setNotifications] = useState([])
+  const [invisibleComponent, setInvisibleComponent] = useState(false)
+  useEffect(()=>{
+    if(socket){
+      socket.on("getNotification", data=>{
+        setNotifications((prev)=>[...prev, data])
+      })
+    }
+  },[socket])
+  const displayNotification = ({senderName, type})=>{
+    let action;
+
+    if (type===1){
+      action="favorited"
```

Comment:
Directly compare between string could be dangerous (we all have typos right?) and can't be detected by typechecker. Let's convert to a enum instead.

Reply:
It's an assignment and doesn't require user input. It's so that I don't have to repeat the action words in the notification message.